build/targets.mk

```diff
@@ -668,11 +668,9 @@ ifeq ($(TARGET),ANDROID)
     TARGET_LDFLAGS += -Wl,--no-warn-mismatch
   endif
 
-  # clang as linker driver adds the option '-pie' to the linker command for the X64 platform.
-  # This option which is incompatible with the option '-shared'.
-  ifeq ($(X64),y)
-    TARGET_LDFLAGS += -no-pie
-  endif
+  # clang as linker driver adds the option '-pie' to the linker command.
+  # This option is incompatible with the option '-shared'.
+  TARGET_LDFLAGS += -no-pie
```

Comment:
There's no explanation why this is suddenly necessary.

Reply:
With the most recent NDK V19.2.5435600 all targets including the 32 bit targets ANDROID7 and ANDROID86 report the error that the parameters -shared and -pie are incompatible.

Here the output from make TARGET=ANDROID7:
  LINK    output/ANDROID7/dbg/bin/libxcsoar-ns.so
/home/hor/Android/ndk-bundle/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin/ld: fatal error: -shared and -pie are incompatible
clang++: error: linker command failed with exit code 1 (use -v to see invocation)

Reply:
I don't see this message with NDK r19c (the latest release). What is this version number you posted?